src/addrman.h

```diff
@@ -410,9 +413,20 @@ friend class CAddrManTest;
 
         OverrideStream<Stream> s(&s_, s_.GetType(), stream_version);
 
-        unsigned char nKeySize;
-        s >> nKeySize;
-        if (nKeySize != 32) throw std::ios_base::failure("Incorrect keysize in addrman deserialization");
+        //! The maximum format this software can Unserialize().
+        static constexpr Format maximum_supported_format = Format::V3_BIP155;
```

Comment:
This is a very badly named constant! The whole point of this PR is that we want the code to be forward compatible if possible, and only in exceptional circumstances make format changes that aren't. If the format is bumped, but lowest_compatible is not, then older software will still be able to read it.

I'd suggest moving this to a class-level static constant called FILE_FORMAT or similar, and use it in both the serialize and unserialize functions.

Reply:
Renamed to `maximum_known_format`.

There is no use for such a constant in `Serialize()`. It is only necessary in `Unserialize()`, thus I think it is better to have it as local variable in `Unserialize()` in order to minimize its span.

Reply:
My point is that the "maximum known format" should always the same as the value that is serialized into the format version byte. It's the format that this software uses.

Defining a constant here means that it needs to be updated at the same time as line 351 where the version is serialized into the file.

Reply:
I see you point now. Moved as a member variable of the class and renamed to `FILE_FORMAT`.